query-engine/query-engine-napi/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+use engine::QueryEngine;
+use napi::{CallContext, Env, JsObject, JsString, JsUndefined, Property};
+use napi_derive::{js_function, module_exports};
+use query_core::QueryExecutor;
+
+mod engine;
+mod error;
+mod exec_loader;
+
+pub(crate) type Result<T> = std::result::Result<T, error::ApiError>;
+pub(crate) type Executor = Box<dyn QueryExecutor + Send + Sync>;
+
+#[js_function(1)]
+fn engine_constructor(ctx: CallContext) -> napi::Result<JsUndefined> {
+    let url = ctx.get::<JsString>(0)?.into_utf8()?;
+
+    let mut this: JsObject = ctx.this_unchecked();
+    let engine = QueryEngine::new(url.as_str()?)?;
+
+    ctx.env.wrap(&mut this, engine)?;
+    ctx.env.get_undefined()
+}
+
+#[js_function(0)]
+fn engine_enable_raw_queries(ctx: CallContext) -> napi::Result<JsUndefined> {
+    let this: JsObject = ctx.this_unchecked();
+
+    let engine: &mut QueryEngine = ctx.env.unwrap(&this)?;
+    engine.enable_raw_queries()?;
+
+    ctx.env.get_undefined()
+}
+
+#[js_function(0)]
+fn engine_connect(ctx: CallContext) -> napi::Result<JsObject> {
+    let this: JsObject = ctx.this_unchecked();
+    let engine: &QueryEngine = ctx.env.unwrap(&this)?;
+    let mut engine: QueryEngine = engine.clone();
+
+    ctx.env
+        .execute_tokio_future(async move { Ok(engine.connect().await?) }, |&mut env, ()| {
+            env.get_undefined()
+        })
+}
+
+#[js_function(0)]
```

Comment:
```suggestion
#[js_function(1)]
```

This takes one argument. It is not working otherwise.

Reply:
Hmm, so `self` counts as a parameter here? Does the `query` work with `1` or should we have `2` parameters there, first being `self` and the second `query`?

Reply:
Whoops.